opensquirrel/circuit_builder.py

```diff
@@ -11,12 +11,15 @@
 from opensquirrel.circuit import Circuit
 from opensquirrel.default_gates import default_gate_aliases, default_gate_set
 from opensquirrel.default_measurements import default_measurement_set
-from opensquirrel.instruction_library import GateLibrary, MeasurementLibrary
-from opensquirrel.ir import IR, Comment, Gate, Measure
-from opensquirrel.register_manager import BitRegister, QubitRegister, RegisterManager
+from opensquirrel.default_resets import default_reset_set
+from opensquirrel.instruction_library import (GateLibrary, MeasurementLibrary,
+                                              ResetLibrary)
+from opensquirrel.ir import IR, Comment, Gate, Measure, Reset
+from opensquirrel.register_manager import (BitRegister, QubitRegister,
+                                           RegisterManager)
 
 
-class CircuitBuilder(GateLibrary, MeasurementLibrary):
+class CircuitBuilder(GateLibrary, MeasurementLibrary, ResetLibrary):
```

Comment:
This should accept `InstructionLibrary` class. This means that the `InstructionLibrary` class has to be refactored. Or else this list will continue growing, i.e `BarrierLibrary, WireCutLibrary, CVLibrary, PulseLibrary`.

Reply:
Add a JIRA ticket and link it to studying the design?